### PR TITLE
Implement From LibAFL Error for Qemu Error

### PR DESCRIPTION
**Pull request by rmalmain**

### CODE REVIEW

File: libafl_qemu/src/qemu/mod.rs [hunk 1]
```diff
@@ -57,6 +57,18 @@ pub enum QemuError {
     RW(QemuRWError),
 }
 
+impl From<QemuError> for libafl::Error {
+    fn from(qemu_error: QemuError) -> Self {
+        libafl::Error::unknown(qemu_error)
```
Comment (domenukk) [inline libafl_qemu/src/qemu/mod.rs:62]: There are things like `TooManyArgs` that sound like they map to `Error::illegal_argument` for examle

**Comment by domenukk** [comment]
So if you're lazy we can merge this, but it might be better to find more useful errors than `unknown` (in almost every case)

**Comment by domenukk** [comment]
Even if we just add a special `qemu` or `instrumentation` error type

**Comment by rmalmain** [comment]
i was thinking of adding an error like instrumentation.
i don't think anything else fits too well

**Comment by rmalmain** [comment]
does it look ok ilke this?

**Comment by rmalmain** [comment]
from libafl's point of view i'd say yes :)
instrumentation sounds ok though, but i thought having a general error for specific things calling into libafl would be convenient at some point. 
suggestions welcome

**Comment by domenukk** [comment]
From LibAFL's point of view the frontend is the UI, the instrumentation is the backend?
Why not Instrumentation or Runtime or something?

File: libafl_bolts/src/lib.rs [hunk 2]
```diff
@@ -300,6 +300,8 @@ pub enum Error {
     Unknown(String, ErrorBacktrace),
     /// Error with the corpora
     InvalidCorpus(String, ErrorBacktrace),
+    /// Error specific to a frontend like QEMU or Frida
+    Frontend(String, ErrorBacktrace),
```
Comment (domenukk) [inline libafl_bolts/src/lib.rs:304]: Frontend just doesn't make sense to me.
Runtime?

**Pull request merged**